tests/io_components/test_write.py

```diff
@@ -117,3 +118,33 @@ def test_raise_if_dataset_exists(store_factory, dataset_function):
     raise_if_dataset_exists(dataset_uuid="ThisDoesNotExist", store=store_factory)
     with pytest.raises(RuntimeError):
         raise_if_dataset_exists(dataset_uuid=dataset_function.uuid, store=store_factory)
+
+
+@pytest.mark.skipif(
+    version.parse(pd.__version__) < version.parse("2"),
+    reason="Timestamp unit coercion is only relevant in pandas >= 2",
+)
+def test_coerce_schema_timestamp_units(store):
+    date = pd.Timestamp(2000, 1, 1)
+
+    mps = [
+        MetaPartition(label="one", data=pd.DataFrame({"a": date, "b": [date]})),
+        MetaPartition(
+            label="two",
+            data=pd.DataFrame({"a": date.as_unit("ns"), "b": [date.as_unit("ns")]}),
+        ),
+    ]
+
+    try:
```

Comment:
The try...except is a bit superfluous, isn't it? Also, not clear why you would only catch the `ValueError`. Maybe just keep `store_dataset_from_partitions(...)`? I think the comment makes it clear what you're trying to accomplish.

To maybe make the test a bit stricter (going beyond "does the code run?") can you read the dataset and check that the values are correct?

Reply:
The reason I added the try catch was because I wanted to turn the `Error` into a `Failure` in pytest for that specific type of exception.

But definitely I think reading the dataset and checking that the values are correct sounds like a better test, I'll go ahead and do that.

Reply:
@jtilly I beefed up the test a little and removed the try except, is this better?

Reply:
Yes, perfect, thank you!